### PR TITLE
Fix #3343: Locate modules with Yarn PnP in PnP projects

### LINKED ISSUE #3343: flow-typed install fails when run on a package using Yarn PnP
I have a project set up using [Yarn PnP][1] and Flow. When I attempt to run `flow-typed install` I get the following error:

```
UNCAUGHT ERROR: Error: Unable to find `node_module/<module>` install directory! Did you forget to run `npm install` before running `flow-typed install`?
    at _callee$ (~/.cache/yarn/v4/npm-flow-typed-2.5.2-967e83c762a1cbfcd52eebaece1ade77944f1714/node_modules/flow-typed/dist/lib/stubUtils.js:103:19)
    at tryCatch (~/.cache/yarn/v4/npm-regenerator-runtime-0.13.2-32e59c9a6fb9b1a4aff09b4930ca2d4477343447/node_modules/regenerator-runtime/runtime.js:45:40)
    at Generator.invoke [as _invoke] (~/.cache/yarn/v4/npm-regenerator-runtime-0.13.2-32e59c9a6fb9b1a4aff09b4930ca2d4477343447/node_modules/regenerator-runtime/runtime.js:271:22)
    at Generator.prototype.(anonymous function) [as next] (~/.cache/yarn/v4/npm-regenerator-runtime-0.13.2-32e59c9a6fb9b1a4aff09b4930ca2d4477343447/node_modules/regenerator-runtime/runtime.js:97:21)
    at asyncGeneratorStep (~/.cache/yarn/v4/npm-flow-typed-2.5.2-967e83c762a1cbfcd52eebaece1ade77944f1714/node_modules/flow-typed/dist/lib/stubUtils.js:38:103)
    at _next (~/.cache/yarn/v4/npm-flow-typed-2.5.2-967e83c762a1cbfcd52eebaece1ade77944f1714/node_modules/flow-typed/dist/lib/stubUtils.js:40:194)
```

In 'PnP' mode, Yarn doesn't copy packages to the `node_modules` directory - it modifies the Node resolver at runtime instead, so flow-typed shouldn't assume it can find them there.

[1]: https://github.com/yarnpkg/rfcs/blob/master/accepted/0000-plug-an-play.md

### PR DESCRIPTION
**Pull request by rtsao**
Resolves #3343

Based on original PR https://github.com/flow-typed/flow-typed/pull/3344

This adds support for resolving dependencies installed with PnP and adds a regression test

### CODE REVIEW

**Review by AndrewSouthpaw — APPROVED** [review]
LGTM. I'm not familiar with PnP but it seems reasonable. It'd be nice to get a second pair of eyes though, @pascalduez ?

File: cli/src/lib/npm/npmProjectUtils.js [hunk 1]
```diff
@@ -25,6 +27,10 @@ type PkgJson = {|
   },
 |};
 
+export type PnpResolver = {
+  resolveToUnqualified: (string, string) => string | null,
+};
```
Comment (pascalduez) [inline cli/src/lib/npm/npmProjectUtils.js:32]:
```suggestion
export type PnpResolver = {|
  resolveToUnqualified: (string, string) => string | null,
|};
```

**Review by pascalduez — APPROVED** [review]

**Comment by AndrewSouthpaw** [comment]
Thanks for seeing it through to the end, @rtsao !

**Comment by GAntoine** [comment]
Looks like this needs a release, I'll get a minor version out this week.

**Pull request merged**